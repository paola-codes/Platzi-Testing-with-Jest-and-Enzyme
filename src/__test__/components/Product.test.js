import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import ProductMock from '../../__mocks__/ProductMock';
import Product from '../../components/Product';

describe('<Product />', () => {
  test('Render of Product component', () => {
    const product = shallow(
      <ProviderMock>
        <Product />
      </ProviderMock>,
    );
    expect(product.length).toEqual(1);
  });
  test('Verify button to buy', () => {
    const handleAddToCart = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <Product
          product={ProductMock}
          handleAddToCart={handleAddToCart}
        />
      </ProviderMock>,
    );
    wrapper.find('button').simulate('click');
    expect(handleAddToCart).toHaveBeenCalledTimes(1);
  });
});

describe('Product Snapshot', () => {
  test('Verify UI of Product component', () => {
    const handleAddToCart = jest.fn();
    const component = renderer.create(
      <ProviderMock>
        <Product
          product={ProductMock}
          handleAddToCart={handleAddToCart}
        />
      </ProviderMock>,
    );
    const product = component.toJSON();
    expect(product).toMatchSnapshot();
  });
});
