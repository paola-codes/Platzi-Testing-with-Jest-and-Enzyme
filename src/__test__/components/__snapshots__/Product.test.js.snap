// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Snapshot Verify UI of Product component 1`] = `
<div
  className="Products-item"
>
  <img
    alt="Camiseta"
    src="https://arepa.s3.amazonaws.com/camiseta.png"
  />
  <div
    className="Products-item-info"
  >
    <h2>
      Camiseta
      <span>
        $
        25
      </span>
    </h2>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit
    </p>
  </div>
  <button
    type="button"
  >
    Comprar
  </button>
</div>
`;
